### PR TITLE
Upgrade svelte-kit to 1.0.1.

### PR DESCRIPTION
**Pull request by tomprince**
I've been a bit confused looking at this project, vs the documentation for sveltekit. It turns out that there was a breaking change, that changed how projects are laid out. There is a migration guide [here](https://github.com/sveltejs/kit/discussions/5774) and a discussion of the motivation [here](https://github.com/sveltejs/kit/discussions/5748).

Note, however, that the migration tool doesn't work particularly well on the project, since before this breaking change, every `.svelte` file in `src/routes` was treated as a page (for example, https://www.spiritislandbuilder.com/spirit-board is a page, which if you look at the console, always has an error). After this change, only the top-level component is actually treated as a page and is directly accessible.

### CODE REVIEW

**Comment by neubee** [comment]
I've been avoiding this one for a while 😅. Breaking changes are such a pain.

**Comment by tomprince** [comment]
> I've been avoiding this one for a while 😅. Breaking changes are such a pain.

Well, it turns out for this project, the change involved renaming two files (and fixing up some comments).

**Comment by tomprince** [comment]
@neubee Is there anything blocking this from landing? (The linting errors are from dev, and are fixed in #155)

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+16
```
Comment (tomprince) [inline .nvmrc:1]: @neubee I added this file to be (somewhat) explicit about the version of node used by cloudflare. I'm guessing you currently have `NODE_VERSION` set in the cloudflare configuration, which has the same effect. Note that svelte-kit 1.0.1 explicitly doesn't support node 17 (as it isn't an LTS version) and cloudflare pages doesn't support node 18.

**Comment by tomprince** [comment]
This branch is deployed [here](https://upgrade-sveltekit.spirit-island-builder-c7s.pages.dev/)

File: .nvmrc (hunk 1 above)
Comment (neubee) [reply .nvmrc:1]: Nice, I did not know this was something you could do.

**Pull request merged**